rules/terraformrules/terraform_naming_convention.go

```diff
@@ -0,0 +1,304 @@
+	Regexp        *regexp.Regexp
+}
+
+// NewTerraformNamingConventionRule returns new rule with default attributes
+func NewTerraformNamingConventionRule() *TerraformNamingConventionRule {
+	return &TerraformNamingConventionRule{}
+}
+
+// Name returns the rule name
+func (r *TerraformNamingConventionRule) Name() string {
+	return "terraform_naming_convention"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *TerraformNamingConventionRule) Enabled() bool {
+	return false
+}
+
+// Severity returns the rule severity
+func (r *TerraformNamingConventionRule) Severity() string {
+	return tflint.NOTICE
+}
+
+// Link returns the rule reference link
+func (r *TerraformNamingConventionRule) Link() string {
+	return tflint.ReferenceLink(r.Name())
+}
+
+// Check checks whether blocks follow naming convention
+func (r *TerraformNamingConventionRule) Check(runner *tflint.Runner) error {
+	log.Printf("[TRACE] Check `%s` rule for `%s` runner", r.Name(), runner.TFConfigPath())
+
+	config := terraformNamingConventionRuleConfig{}
+	config.Format = "snake_case"
+	if err := runner.DecodeRuleConfig(r.Name(), &config); err != nil {
+		return err
+	}
+
+	defaultNameValidator, err := config.getNameValidator()
+	if err != nil {
+		return fmt.Errorf("Invalid default configuration: %v", err)
+	}
+
+	dataNameValidator := defaultNameValidator
+	if config.Data != nil {
+		nameValidator, err := config.Data.getNameValidator()
+		if err != nil {
+			return fmt.Errorf("Invalid data configuration: %v", err)
+		}
+
+		dataNameValidator = nameValidator
+	}
+
+	localsNameValidator := defaultNameValidator
+	if config.Locals != nil {
+		nameValidator, err := config.Locals.getNameValidator()
+		if err != nil {
+			return fmt.Errorf("Invalid locals configuration: %v", err)
+		}
+
+		localsNameValidator = nameValidator
+	}
+
+	moduleNameValidator := defaultNameValidator
+	if config.Module != nil {
+		nameValidator, err := config.Module.getNameValidator()
+		if err != nil {
+			return fmt.Errorf("Invalid module configuration: %v", err)
+		}
+
+		moduleNameValidator = nameValidator
+	}
+
+	outputNameValidator := defaultNameValidator
+	if config.Output != nil {
+		nameValidator, err := config.Output.getNameValidator()
+		if err != nil {
+			return fmt.Errorf("Invalid output configuration: %v", err)
+		}
+
+		outputNameValidator = nameValidator
+	}
+
+	resourceNameValidator := defaultNameValidator
+	if config.Resource != nil {
+		nameValidator, err := config.Resource.getNameValidator()
+		if err != nil {
+			return fmt.Errorf("Invalid resource configuration: %v", err)
+		}
+
+		resourceNameValidator = nameValidator
+	}
+
+	variableNameValidator := defaultNameValidator
+	if config.Variable != nil {
+		nameValidator, err := config.Variable.getNameValidator()
+		if err != nil {
+			return fmt.Errorf("Invalid variable configuration: %v", err)
+		}
+
+		variableNameValidator = nameValidator
+	}
+
+	// Actually run any checks for modules
+	if dataNameValidator != nil {
```

Comment:
I might make sense to move the logic if handling each of the block types into their own methods

Reply:
Thanks for the suggestion, I moved the code to separate methods. I'd like to make things a bit more concise, as there's a fair amount of duplication, but I don't think my go abilities & available time allow for that.